packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -1591,6 +1591,53 @@ abstract class _RenderPhysicalModelBase<T> extends _RenderCustomClip<T> {
     description.add(DiagnosticsProperty<Color>('color', color));
     description.add(DiagnosticsProperty<Color>('shadowColor', color));
   }
+
+  @protected
+  Path get clipPath;
+
+  @override
+  @mustCallSuper
+  void paint(PaintingContext context, Offset offset) {
+    assert(() {
+      RenderObject ancestor = parent;
+      while (ancestor != null) {
+        if (ancestor is _RenderPhysicalModelBase) {
+          // We're the direct child of another physical model. That is the one
+          // that needs to be checked, as it has established a new elevation
+          // context for us.
+          return true;
+        }
+        if (ancestor is RenderOffstage) {
+          // Keep the RenderOffstage to pass to the check, which (at least for
+          // now) will scope checks to children of the same RenderOffstage.
+          // This is to prevent getting lots of noise right now during route
+          // transitions, and is probably not an entirely valid way of checking
+          // this.  Need more input to determine the right way to handle
+          // transitions involving elevations.
+          break;
+        }
+        ancestor = ancestor.parent;
+      }
+      final Path diff = owner.debugCheckElevationData(
+        elevation,
+        Rect.fromLTWH(offset.dx, offset.dy, size.width, size.height),
+        clipPath,
+        this,
+        ancestor,
+      );
+      if (diff != null) {
+        final Matrix4 matrix = Matrix4.identity()..translate(offset.dx, offset.dy);
```

Comment:
Is the offset the only thing that can change where to paint the path? What about Transformation widgets around this one? Probably worth adding a test for such a case.

And maybe, can we just use `applyPaintTransform` here?